pretrain_gpt_alcf.py

```diff
@@ -434,18 +434,42 @@ def git_ds_info():
 def main():
     # if RANK == 0:
     #     setup_wandb()
-    from torch.profiler import profile, record_function, ProfilerActivity
-    # with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA]) as prof:
-    model = pretrain(
-        train_valid_test_datasets_provider,
-        model_provider,
-        ModelType.encoder_or_decoder,
-        forward_step,
-        args_defaults={'tokenizer_type': 'GPT2BPETokenizer'},
-        data_post_process=data_post_process
-    )
-    # args = get_args()
-    # prof.export_chrome_trace(f"{args.tensorboard_dir}/torch-trace-{RANK}-of-{WORLD_SIZE}.json")
+    import socket
+    from mpi4py import MPI
+    rank = MPI.COMM_WORLD.rank
+
+    if rank == 0:
+        master_addr = socket.gethostname()
+    else:
+        master_addr = None
+
+        master_addr = MPI.COMM_WORLD.bcast(master_addr, root=0)
+    os.environ["MASTER_ADDR"] = master_addr
+    os.environ["MASTER_PORT"] = str(2345)
```

Comment:
@zhenghh04 just a thought, but could there be a potential issue when setting

```python
os.environ["MASTER_PORT"] = str(2345)
```

if DDP (/ DeepSpeed) are expecting (/ using ?) a different `MASTER_PORT` ?  

i.e. I believe for PyTorch the default is `29500`

Reply:
Now that I'm thinking about it a bit, I've got similar logic implemented here in this 

[`ezpz.dist.setup_torch_DDP(...)`](https://github.com/saforem2/ezpz/blob/main/src/ezpz/dist.py#L471)

method from [`ezpz`](https://github.com/saforem2/ezpz.git), but I wonder now if this could be related to the hanging issues we've been experiencing at startup 🤔...